generator/uscore/uscore_generator.rb

```diff
@@ -369,6 +350,63 @@ def param_value_name(param)
         end
       end
 
+      def get_first_search(search_parameters, sequence)
+        search_code = ''
+
+        save_resource_ids_in_bundle_arguments = [
+          "versioned_resource_class('#{sequence[:resource]}')",
+          'reply',
+          validation_profile_uri(sequence)
+        ].compact.join(', ')
+
+        if search_parameters == ['patient'] || sequence[:delayed_sequence] || search_param_constants(search_parameters, sequence)
+          search_code = %(
+            #{get_search_params(search_parameters, sequence)}
+            reply = get_resource_by_params(versioned_resource_class('#{sequence[:resource]}'), search_params)
+            assert_response_ok(reply)
+            assert_bundle_response(reply)
+
+            resource_count = reply&.resource&.entry&.length || 0
+            @resources_found = true if resource_count.positive?
+
+            skip 'No resources appear to be available for this patient. Please use patients with more information.' unless @resources_found
+
+            @#{sequence[:resource].downcase} = reply&.resource&.entry&.first&.resource
+            @#{sequence[:resource].downcase}_ary = fetch_all_bundled_resources(reply&.resource)
+            save_resource_ids_in_bundle(#{save_resource_ids_in_bundle_arguments})
+            save_delayed_sequence_references(@#{sequence[:resource].downcase}_ary)
+            validate_search_reply(versioned_resource_class('#{sequence[:resource]}'), reply, search_params)
+          )
+        else
+          # assume only patient + one other parameter
```

Comment:
This method is pretty big, is it worth breaking out the consequent and alternative logic in the if statement into their own methods?  I don't want this to be a show stopper for the PR if it doesn't make sense though.

Reply:
Moved both to their own function.